### ISSUE TITLE
Keep configuration/version set in components launched/replaced by addons-starter

### ISSUE THREAD

**Issue by wjimenez5271** [rancher/rancher#10045]
When a user upgrades an addon like Tiller after they've deployed K8s on Rancher, it would be useful if Rancher could retain at least that version (or higher) when upgrading. Since some of these changes are difficult for Rancher to detect, a configuration setting might be a good option where the user can indicate the minimum version currently installed. 

Having a way to store it is important rather than just an option in the upgrade workflow as users can forget to set this version at upgrade and cause unexpected problems. Storing the value in Rancher prevents people from forgetting. 

Currently requested:
* Keep Tiller version
* Keep kube-dns configmap data

**Comment by deniseschannon** [comment]
Available with `v1.8.5-rancher1`

**Comment by soumyalj** [comment]
Tested with upgrade from K8s v1.7.7-rancher1 to K8s v1.8.5rancher1.( This is with upgrading rancher from v1.6.10 to v1.6.13-rc1)
In k8s v1.7.7-rancher1, upgraded tiller to the latest version gcr.io/kubernetes-helm/tiller:v2.7.2
Edited the kube-dns cofigmap to include upstreamNameservers as below:
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: kube-dns
  namespace: kube-system
data:
  upstreamNameservers: |
    ["172.16.0.1"]

```
Upgraded rancher from v1.6.10 to v1.6.10-rc1. Upgraded Kubernetes to K8s v1.8.5-rancher1.
After upgrading Kubernetes, verified that tiller still had the  latest version gcr.io/kubernetes-helm/tiller:v2.7.2
```
Containers:
  tiller:
    Container ID:	docker://a7d13ae36180b97cb17e904298ab713f63aacc81b5930332c8f596e420743e10
    Image:		gcr.io/kubernetes-helm/tiller:v2.7.2
    Image ID:		docker-pullable://gcr.io/kubernetes-helm/tiller@sha256:df7f227fa722afc4931c912c1cad2c47856ec94f4d052ccceebcb16dd483dad8

```
Kube-dns configmap also had the updated information
```
kubectl describe configmap kube-dns --namespace kube-system
Name:		kube-dns
Namespace:	kube-system
Labels:		addonmanager.kubernetes.io/mode=EnsureExists
Annotations:	kubectl.kubernetes.io/last-applied-configuration={"apiVersion":"v1","kind":"ConfigMap","metadata":{"annotations":{},"labels":{"addonmanager.kubernetes.io/mode":"EnsureExists"},"name":"kube-dns","names...

Data
====
upstreamNameservers:
----
["8.8.8.8"]

Events:	<none>
```
